plugin/contrib/mesos/pkg/executor/executor_test.go

```diff
@@ -195,3 +214,363 @@ func TestSuicide_WithTasks(t *testing.T) {
+		if ok {
+			if reflect.DeepEqual(initialPodUpdate, update) {
+				receivedInitialPodUpdate = true
+			}
+		}
+	case <-time.After(50 * time.Millisecond):
+	}
+	assert.Equal(t, true, receivedInitialPodUpdate,
+		"executor should have sent an initial PodUpdate "+
+			"to the updates chan upon registration")
+
+	assert.Equal(t, true, executor.isConnected(), "executor should be connected")
+}
+
+// TestExecutorDisconnect ensures that the executor thinks that it is not
+// connected after a call to Disconnected has occured.
+func TestExecutorDisconnect(t *testing.T) {
+	mockDriver := MockExecutorDriver{}
+	executor := NewTestKubernetesExecutor()
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+	executor.Disconnected(mockDriver)
+
+	assert.Equal(t, false, executor.isConnected(),
+		"executor should not be connected after Disconnected")
+}
+
+// TestExecutorReregister ensures that the executor thinks it is connected
+// after a connection problem happens, followed by a call to Reregistered.
+func TestExecutorReregister(t *testing.T) {
+	mockDriver := MockExecutorDriver{}
+	executor := NewTestKubernetesExecutor()
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+	executor.Disconnected(mockDriver)
+	executor.Reregistered(mockDriver, nil)
+
+	assert.Equal(t, true, executor.isConnected(), "executor should be connected")
+}
+
+// TestExecutorLaunchAndKillTask ensures that the executor is able to launch
+// and kill tasks while properly bookkeping its tasks.
+func TestExecutorLaunchAndKillTask(t *testing.T) {
+	// create a fake pod watch. We use that below to submit new pods to the scheduler
+	podListWatch := NewMockPodsListWatch(api.PodList{})
+
+	// create fake apiserver
+	testApiServer := NewTestServer(t, api.NamespaceDefault, &podListWatch.list)
+	defer testApiServer.server.Close()
+
+	mockDriver := MockExecutorDriver{}
+	updates := make(chan interface{}, 1024)
+	config := Config{
+		Docker:  dockertools.ConnectToDockerOrDie("fake://"),
+		Updates: updates,
+		APIClient: client.NewOrDie(&client.Config{
+			Host:    testApiServer.server.URL,
+			Version: testapi.Version(),
+		}),
+		Kubelet: &kubelet.Kubelet{},
+		PodStatusFunc: func(kl *kubelet.Kubelet, pod *api.Pod) (api.PodStatus, error) {
+			return api.PodStatus{
+				ContainerStatuses: []api.ContainerStatus{
+					{
+						Name: "foo",
+						State: api.ContainerState{
+							Running: &api.ContainerStateRunning{},
+						},
+					},
+				},
+				Phase: api.PodRunning,
+			}, nil
+		},
+	}
+	executor := New(config)
+
+	executor.Init(mockDriver)
+	executor.Registered(mockDriver, nil, nil, nil)
+
+	gotInitialUpdate := false
+	select {
+	case <-updates:
+		gotInitialUpdate = true
+	case <-time.After(5 * time.Millisecond):
+	}
+	assert.Equal(t, true, gotInitialUpdate,
+		"Executor should send an initial update on Registration")
+
+	pod := newTestPod(1)
+	podTask, err := podtask.New(api.NewDefaultContext(), "default/pod1",
```

Comment:
`default/pod1` is confusing for a podtask id because it looks like a namespace/name construct. the scheduler implementation uses "" here instead and lets podtask.New generate an ID. we could do that, or else specify something simpler like "foo" or "bar"